spanner/session_test.go

```diff
@@ -1653,6 +1679,82 @@ func TestMaintainer(t *testing.T) {
 	})
 }
 
+func TestMultiplexSessionWorker(t *testing.T) {
+	t.Parallel()
+	if !isMultiplexEnabled {
+		t.Skip("Multiplexing is not enabled")
+	}
+	ctx := context.Background()
+
+	server, client, teardown := setupMockedTestServerWithConfig(t,
+		ClientConfig{
+			SessionPoolConfig: SessionPoolConfig{
+				MultiplexSessionCheckInterval: time.Millisecond,
+			},
+		})
+	defer teardown()
+	_, err := client.Single().ReadRow(ctx, "Albums", Key{"foo"}, []string{"SingerId", "AlbumId", "AlbumTitle"})
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	sp := client.idleSessions
+	waitFor(t, func() error {
+		sp.mu.Lock()
+		defer sp.mu.Unlock()
+		if sp.multiplexedSession == nil {
+			return errInvalidSessionPool
+		}
+		return nil
+	})
+	if !testEqual(uint(1), server.TestSpanner.TotalSessionsCreated()) {
+		t.Fatalf("expected 1 session to be created, got %v", server.TestSpanner.TotalSessionsCreated())
+	}
+	// Will cause session creation RPC to be fail.
+	server.TestSpanner.PutExecutionTime(MethodCreateSession,
+		SimulatedExecutionTime{
+			Errors:    []error{status.Errorf(codes.Unavailable, "try later")},
```

Comment:
Unavailable will be retried by Gax, so it won't bubble up to the client library. Instead, it will just seem like it takes a longer time. So I think that the error code that is used by this test should be something else.